### ISSUE TITLE
Android support?

### ISSUE THREAD

**Issue by adorokhine** [JetBrains/skiko#423]
Will there be support for using the skiko API on Android? That would make it much simpler to write cross-platform apps/rendering.

**Comment by olonho** [comment]
Makes sense.

**Comment by olonho** [comment]
https://github.com/JetBrains/skiko/pull/425 provides basic build support, check with

```console
./gradlew -Pskiko.android.enabled=true skikoJvmRuntimeJar-android-arm64
./gradlew -Pskiko.android.enabled=true skikoJvmRuntimeJar-android-x64
```

NDK have to be installed.

**Comment by jimgoog** [comment]
@adorokhine What kind of xplat apps are you trying to build, and why is Compose MPP not adequate for your use case?

**Comment by adorokhine** [comment]
This particular app is a game. On Desktop it draws onto a Swing Canvas, on Android it draws onto a SurfaceView, and in JS it draws onto an HTML5 canvas. Haven't looked at iPhone yet but I imagine there's some way to get a GL surface going there too.

I haven't taken a close look at Compose MPP but since the app doesn't really have any widgets except one single canvas, and considering that it also has to run on desktop, it doesn't seem like the best fit for my usecase.

**Comment by jimgoog** [comment]
> the app doesn't really have any widgets except one single canvas

Compose MPP does provide a canvas widget, so your app could just use Compose to open a giant Canvas and then you can paint to your heart's content.  If (as your game grows) you ever decide that you want some additional UI to display in-game menus or dialogs or any other UI, you would have the full power of Compose UI available to you.

> and considering that it also has to run on desktop, it doesn't seem like the best fit for my usecase.

Compose MPP does run on Desktop.  In fact, Desktop (Windows+Mac+Linux) is one of the best supported platforms, after Android.  Full support for canvas widgets on other platforms is still in the works but expected soon.  Skiko is a prerequisite for Compose and Compose is a motivating use case for Skiko so the two tend to be in fairly tight in terms of stability and completeness on any given platform.

> I haven't taken a close look at Compose MPP

I'd recommend taking a look.  It's pretty cool.

**Comment by thaihuynhxyz** [comment]
Hope we will have `SkiaLayer.attachTo(surface: Surface)` or `SkiaLayer.attachTo(holder: SurfaceHolder)`.

**Comment by olonho** [comment]
Android support is available starting with Skiko 0.7.7, please take a look.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/SKIKO-423">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
